javaslang/src/main/java/javaslang/collection/PriorityQueue.java

```diff
@@ -0,0 +1,826 @@
+    public PriorityQueue<T> peek(Consumer<? super T> action) {
+        Objects.requireNonNull(action, "action is null");
+        if (!isEmpty()) {
+            action.accept(head());
+        }
+        return this;
+    }
+
+    @Override
+    public PriorityQueue<T> replace(T currentElement, T newElement) {
+        Objects.requireNonNull(currentElement, "currentElement is null");
+        Objects.requireNonNull(newElement, "newElement is null");
+        return ofAll(comparator, toList().replace(currentElement, newElement));
+    }
+
+    @Override
+    public PriorityQueue<T> replaceAll(T currentElement, T newElement) {
+        Objects.requireNonNull(currentElement, "currentElement is null");
+        Objects.requireNonNull(newElement, "newElement is null");
+        return ofAll(comparator, toList().replaceAll(currentElement, newElement));
+    }
+
+    @Override
+    public PriorityQueue<T> retainAll(Iterable<? extends T> elements) {
+        return Collections.retainAll(this, elements);
+    }
+
+    public PriorityQueue<T> removeAll(Iterable<? extends T> elements) {
+        return Collections.removeAll(this, elements);
+    }
+
+    @Override
+    public PriorityQueue<T> scan(T zero, BiFunction<? super T, ? super T, ? extends T> operation) {
+        return Collections.scanLeft(this, zero, operation, PriorityQueue.empty(comparator), PriorityQueue::enqueue, Function.identity());
+    }
+
+    @Override
+    public <U> PriorityQueue<U> scanLeft(U zero, BiFunction<? super U, ? super T, ? extends U> operation) {
+        Objects.requireNonNull(operation, "operation is null");
+        return Collections.scanLeft(this, zero, operation, empty(naturalComparator()), PriorityQueue::enqueue, Function.identity());
+    }
+
+    @Override
+    public <U> PriorityQueue<U> scanRight(U zero, BiFunction<? super T, ? super U, ? extends U> operation) {
+        Objects.requireNonNull(operation, "operation is null");
+        return Collections.scanRight(this, zero, operation, empty(naturalComparator()), PriorityQueue::enqueue, Function.identity());
+    }
+
+    @Override
+    public Iterator<? extends PriorityQueue<T>> sliding(long size) {
+        return iterator().sliding(size).map(v -> ofAll(comparator, v));
+    }
+
+    @Override
+    public Iterator<? extends PriorityQueue<T>> sliding(long size, long step) {
+        return iterator().sliding(size, step).map(v -> ofAll(comparator, v));
+    }
+
+    @Override
+    public Tuple2<? extends PriorityQueue<T>, ? extends PriorityQueue<T>> span(Predicate<? super T> predicate) {
+        Objects.requireNonNull(predicate, "predicate is null");
+        return Tuple.of(takeWhile(predicate), dropWhile(predicate));
+    }
+
+    @Override
+    public Option<? extends PriorityQueue<T>> tailOption() {
+        return isEmpty() ? Option.none() : Option.some(tail());
+    }
+
+    @Override
+    public PriorityQueue<T> take(long n) {
+        if (n >= length()) {
+            return this;
+        } else {
```

Comment:
You will receive my review this weekend. I had a hard working week and need time to dive into the code...